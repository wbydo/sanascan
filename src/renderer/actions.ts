import * as types from "./types";

export interface Action {
  type: string;
}

export const increment: () => Action = () => {
  return {
    type: types.INCREMENT,
  };
};
