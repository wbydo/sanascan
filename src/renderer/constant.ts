export const chars: string[][] = [
  ["あ", "か", "さ", "た", "な", "は", "ま", "や", "ら", "わ"],
  ["い", "き", "し", "ち", "に", "ひ", "み", "",   "り", ""],
  ["う", "く", "す", "つ", "ぬ", "ふ", "む", "ゆ", "る", "を"],
  ["え", "け", "せ", "て", "ね", "へ", "め", "",   "れ", ""],
  ["お", "こ", "そ", "と", "の", "ほ", "も", "よ", "ろ", "ん"],
];

export const MAX_COLUMN_INDEX = chars[0].length - 1;
export const MAX_ROW_INDEX = chars.length - 1;

export const url = "http://localhost:8000";
