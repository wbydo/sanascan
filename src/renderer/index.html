<!DOCTYPE html>
<html lang="ja">
<head>
  <meta charset="utf-8">
  <title>YACOS</title>
  <link rel="stylesheet" href="css/style.css">
</head>
<body>
  <div id="root"></div>
</body>
</html>
