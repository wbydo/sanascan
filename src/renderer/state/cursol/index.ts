import reducer from "./reducers";

import { increment } from "./actions";

export const cursolActions = {
  increment,
};

export const cursolReducer = reducer;
