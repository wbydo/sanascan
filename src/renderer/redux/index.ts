import { createStore, applyMiddleware } from "redux";
import { combineReducers} from "redux";

import { composeWithDevTools } from "redux-devtools-extension";

import { reducer as configWindowReducer } from "./state/configWindow";
import { reducer as cursolReducer } from "./state/cursol";
import { reducer as developerModeReducer } from "./state/developerMode";
import { reducer as estimatorReducer } from "./state/estimator";
import { reducer as timerReducer } from "./state/timer";

import { middlewares as timerMiddlewares } from "./cross/timer";
import { middlewares as httpMiddlewares } from "./cross/http";

interface CursolProperty {
  activeColumn: number;
  activeRow: number;
}

export type CursolState = (
  {mode: "normal", direction: "column"} & CursolProperty
) | (
  {mode: "normal", direction: "row"} & CursolProperty
) | (
  {mode: "proposal", direction: "column"} & CursolProperty
);

export interface RootState {
  configWindow: {
    isActive: boolean;
    scanSpeed: number;
  };
  cursol: CursolState;
  developerMode: {
    isActive: boolean;
    estimator: boolean;
    timer: boolean;
  };
  timer: {
    isActive: boolean;
    scanSpeed: number;
  };
  estimator: {
    id: number | null;
    result: string;
  };
}

const middlewares = [
  ...httpMiddlewares,
  ...timerMiddlewares,
];

const enhancer = composeWithDevTools(
  applyMiddleware(...middlewares),
);

export const reducer = combineReducers({
  configWindow: configWindowReducer,
  cursol: cursolReducer,
  developerMode: developerModeReducer,
  estimator: estimatorReducer,
  timer: timerReducer,
});

export const store = createStore(
  reducer,
  enhancer,
);

export { operations } from "./operations";
