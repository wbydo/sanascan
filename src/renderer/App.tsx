import * as React from "react";
import {connect} from "react-redux";
import {Dispatch} from "redux";

import CharacterBoard from "./CharacterBoard";
import {Action, asyncIncrement} from "./actions";

interface Props {
  asyncIncrement: () => void;
}

type MapDispatchToProps = (dispatch: Dispatch<Action>) => Props;

class App extends React.Component<Props, {}> {
  public render() {
    return(
      <div>
        <CharacterBoard />
        <button type="button" onClick={this.handleOnClick}>
          async
        </button>
      </div>
    );
  }

  public handleOnClick = (event: React.MouseEvent): void => {
    return this.props.asyncIncrement();
  }
}

const mdp: MapDispatchToProps = (dispatch) => {
  return {
    asyncIncrement: () => dispatch(asyncIncrement()),
  };
};

export default connect(null, mdp)(App);
