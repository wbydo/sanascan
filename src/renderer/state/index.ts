import { createStore, applyMiddleware } from "redux";
import { combineReducers, Dispatch } from "redux";

import { composeWithDevTools } from "redux-devtools-extension";

import { reducer as configWindowReducer } from "./configWindow";
import { operations as configWindowOperations } from "./configWindow";

import { reducer as cursolReducer } from "./cursol";

import { reducer as estimatorReducer } from "./estimator";
import { middleware as estimatorMiddleware } from "./estimator";
import { operations as estimatorOperations } from "./estimator";

import { reducer as timerReducer } from "./timer";
import { middleware as timerMiddleware } from "./timer";

export interface RootState {
  configWindow: {
    isActive: boolean;
    scanSpeed: number;
  };
  cursol: {
    activeColumn: number;
  };
  timer: {
    id: number | null;
    isActive: boolean;
    scanSpeed: number;
  };
  estimator: {
    id: number | null;
    result: string;
  };
}

const middlewares = [
  timerMiddleware,
  estimatorMiddleware,
];

const enhancer = composeWithDevTools(
  applyMiddleware(...middlewares),
);

export const reducer = combineReducers({
  configWindow: configWindowReducer,
  cursol: cursolReducer,
  estimator: estimatorReducer,
  timer: timerReducer,
});

export const store = createStore(
  reducer,
  enhancer,
);

export const operations = (dispatch: Dispatch) => {
  return {
    configureWindowOpen: configWindowOperations.windowOpen(dispatch),
    resetEstimator: estimatorOperations.reset(dispatch),
    sendKey: estimatorOperations.sendKey(dispatch),
    startFetchEstimatorId: estimatorOperations.fetchId(dispatch),
  };
};
