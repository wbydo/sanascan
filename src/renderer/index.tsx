import * as React from "react";
import * as ReactDOM from "react-dom";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import createSagaMiddleware from "redux-saga";

import App from "./App";
import reducer from "./reducers";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  applyMiddleware(sagaMiddleware),
);

sagaMiddleware.run(rootSaga);

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root"),
);
