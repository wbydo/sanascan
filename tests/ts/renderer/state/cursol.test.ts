import snapshotDiff from "snapshot-diff";

import { reducer } from "sanascan/renderer/redux";

import { actions } from "sanascan/renderer/redux/state/cursol";

import { MAX_COLUMN_INDEX } from "sanascan/renderer/constant";

import { initialState } from "./initialState";

test("activeColumnが1増える", () => {
  const result = reducer(
    initialState,
    actions.increment(),
  );

  expect(
    snapshotDiff( initialState, result),
  ).toMatchSnapshot();
});

test("activeColumnが0に戻る", () => {
  const x = MAX_COLUMN_INDEX;
  const state = {
    ...initialState,
    cursol: {
      ...initialState.cursol,
      activeColumn: x,
    },
  };

  const result = reducer(
    state,
    actions.increment(),
  );

  expect(
    snapshotDiff(state, result),
  ).toMatchSnapshot();
});

describe("modeが設定出来る", () => {
  const thisTest = (i: "normal" | "proposed") => {
    test(i + "の場合", () => {
      const result = reducer(
        initialState,
        actions.setMode(i),
      );

      expect(
        snapshotDiff( initialState, result),
      ).toMatchSnapshot();
    });
  };

  const normal = "normal";
  thisTest(normal);

  const proposed = "proposed";
  thisTest(proposed);
});
