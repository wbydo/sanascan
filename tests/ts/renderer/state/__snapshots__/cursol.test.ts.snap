// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeColumnが0に戻る 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,11 @@
    \\"configWindow\\": Object {
      \\"isActive\\": false,
      \\"scanSpeed\\": 0,
    },
    \\"cursol\\": Object {
-     \\"activeColumn\\": 9,
+     \\"activeColumn\\": 0,
      \\"mode\\": \\"proposed\\",
    },
    \\"developerMode\\": true,
    \\"estimator\\": Object {
      \\"id\\": null,"
`;

exports[`activeColumnが1増える 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,11 +2,11 @@
    \\"configWindow\\": Object {
      \\"isActive\\": false,
      \\"scanSpeed\\": 0,
    },
    \\"cursol\\": Object {
-     \\"activeColumn\\": 0,
+     \\"activeColumn\\": 1,
      \\"mode\\": \\"proposed\\",
    },
    \\"developerMode\\": true,
    \\"estimator\\": Object {
      \\"id\\": null,"
`;

exports[`modeが設定出来る normalの場合 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -3,11 +3,11 @@
      \\"isActive\\": false,
      \\"scanSpeed\\": 0,
    },
    \\"cursol\\": Object {
      \\"activeColumn\\": 0,
-     \\"mode\\": \\"proposed\\",
+     \\"mode\\": \\"normal\\",
    },
    \\"developerMode\\": true,
    \\"estimator\\": Object {
      \\"id\\": null,
      \\"result\\": \\"\\","
`;

exports[`modeが設定出来る proposedの場合 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
